.github/workflows/build.yml

```diff
@@ -137,14 +137,14 @@ jobs:
     needs: compiler
     runs-on: ubuntu-20.04
     env:
-      GRADLE_VERSION: "7.6"
+      GRADLE_VERSION: "8.0.2"
     steps:
       - uses: actions/checkout@v3
 
       - uses: actions/setup-java@v3
         with:
           distribution: temurin
-          java-version: 17
+          java-version: 19
```

Comment:
19 is not an LTS version, we should stick with LTS versions?

Reply:
for maintainability i think we should stick with 11 (which is guaranteed using `-release` flag). i trust JDK's capability of down-compiling with newer Java versions.

for build tool chain i think it is okay to move along with newest stable release.

Reply:
the `-release` flag was introduced by @ctubbsii

Reply:
my main concern is that someone accidentally introduced some code that requires feature not available in LTS versions, but if that's already guaranteed by `-release` flag then I think this is fine.

the other concern is that the non-LTS versions all have a relatively much shorter support lifespan so we need to update those to supported versions much more frequently. For example, according to https://www.oracle.com/java/technologies/java-se-support-roadmap.html 19 is no longer supported?

Reply:
we can move the java 21 in September but atm i think it is fine given there's no much of a support needed, given the compiled `.class` files are in Java 11. The only support needed is in the build system in this repo.

Reply:
out of support would also mean that it won't get security fixes, so if a security bug is discovered in jdk 19 now the fix would only land on 20 and LTS versions, not 19, I assume? But maybe Oracle's "sustaining support" means it would still get security fixes?

if we are not using any new features introduced since 17 (the latest LTS version right now), I'm not sure why do we _need to_ use java 19 here?

Reply:
this is to check if we are prepared to run build/tests with upcoming releases.
I would recommend to use 20 here

Reply:
I don't see any reason to test with the newer JDK that's available. But, I would suggest prioritizing testing against a JDK that developers are more likely to be using... like the current LTS. If we want to build against both, it's pretty easy to set up a Matrix build in GitHub Actions.